Automattic/wp-calypso
# Checkout: use useShoppingCart for WPCheckoutOrderSummary

#### Changes proposed in this Pull Request

The `WPCOMCartItem` type was created as a sort of hack to allow passing `ResponseCartItem` object data through checkout via the `useLineItems` hook that's part of the `composite-checkout` package, since the generic product data available in that hook's types is not sufficient. However, now that `useShoppingCart` is available, we can get the `ResponseCartItem` products directly where they are needed. This will make it easier to manipulate and query cart items in the code since there will only be one data type we need to understand, and since `ResponseCartItem` predates `WPCOMCartItem`, there's already considerable support for it in calypso.

This PR replaces `useLineItems` in `WPCheckoutOrderSummary` with `useShoppingCart`.

![sidebar](https://user-images.githubusercontent.com/2036909/107590202-9233b380-6bd5-11eb-82e2-32ee41aa614d.png)

#### Testing instructions

- Visit checkout with a product in your cart.
- Verify that the sidebar (at desktop width) includes the total matching the total from the main column.
- Set the postal code and country to a location that applies taxes (eg: 10001, US).
- Verify that the sidebar includes the taxes matching the taxes from the main column.
- Add a coupon to your cart.
- Verify that the sidebar includes the coupon discount.
